docs/source/faq.rst

```diff
@@ -332,3 +332,45 @@ Note that the above examples are similar to running the garbage collector inside
 
     :class:`~optuna.integration.ChainerMNStudy` does currently not provide ``gc_after_trial`` nor callbacks for :func:`~optuna.integration.ChainerMNStudy.optimize`.
     When using this class, you will have to call the garbage collector inside the objective function.
+
+How can I output a log only when the best value is updated?
+-----------------------------------------------------------
+
+Please turn off the logging feature of optuna and implement your own logging-only callback function.
+The implemented callback can be passed to `study.optimize`.
+The following is an example.
+
+.. code-block:: python
+
+    import optuna
+
+    INF = 1e+9
+
+
+    # Turn off the log of optuna.
+    optuna.logging.set_verbosity(optuna.logging.WARN)
+
+
+    def objective(trial):
+        # Preserve the previous best value before computing objective value.
+        n_completed_trials = len(trial.study.get_trials(states=(optuna.trial.TrialState.COMPLETE,)))
+        if n_completed_trials > 0:
+            trial.set_user_attr("previous_best_value", trial.study.best_value)
```

Comment:
I think it would be better to move this code that the callback depends on into the callback itself, so that the design is more modular.

Change logging_callback to something like:
```
previous_best_value = frozen_trial.user_attrs.get("previous_best_value", None)
if study.best_value != previous_best_value:
    trial.set_user_attr("previous_best_value", trial.study.best_value)
    print stuff
```

Reply:
Unfortunately, in the current Optuna, we cannot overwrite the attributes of the completed Trial, so `frozen_trial.set_user_attr("previous_best_value", trial.study.best_value)` does not work in the callback function.

However, I`ve noticed that we can replace `frozen_trial.set_user_attr("previous_best_value", trial.study.best_value)`  with `study.set_user_attr("previous_best_value", trial.study.best_value)` here. As you suggested, it simplify the example. Thank you for your insightful comments. PTAL.